pkg/router/template/router.go

```diff
@@ -878,6 +878,14 @@ func (r *templateRouter) dynamicallyReplaceEndpoints(id ServiceUnitKey, service
 			weight = 0
 		}
 
+		if cfg.TLSTermination == routev1.TLSTerminationReencrypt && cfg.VerifyServiceHostname && cfg.ActiveServiceUnits > 1 {
```

Comment:
Do you need to check for reencrypt? `cfg.VerifyServiceHostname` is only set to true if it's reencrypt via:
https://github.com/openshift/router/blob/9779c5eed8d4f327e7921eda09f011bc33b3e1c4/pkg/router/template/router.go#L1020-L1022

```suggestion
		if cfg.VerifyServiceHostname && cfg.ActiveServiceUnits > 1 {
```

Reply:
Also, is there any reason this can't be earlier in the function? it's within the for loop of `service.ServiceAliasAssociations`, but doesn't use anything from the iteration.

Reply:
> Right, it's redundant. I just wanted to highlight the fact that it only impacts reencrypt routes.

That's fine.

> It uses cfg (type ServiceAliasConfig) which is retrieved from the state. This loop is supposed to iterate over all the routes associated with a given service. Also, I wanted this condition to be close to the place where we call ReplaceRouteEndpoints of DCM.

Whoops, sorry I missed the `cfg`.